### DISCUSSION TITLE
Rpm package for regolith-desktop

### DISCUSSION THREAD

**Discussion by gogsbread** [regolith-linux/regolith-desktop#734] [General]
#26  Working on building a rpm package for regolith-desktop. To understand the basics I am trying to locally build the regolith-desktop package for debian using `./local-build.sh ../.. ext-debian.sh regolith-desktop https://github.com/regolith-linux/regolith-desktop.git debian-bullseye bullseye`. 
My hope is that once I understand the basics and the github actions flow, I can fill-in details for `ext-rpm.sh` and produce necessary artifacts. Am I going in the right direction?

**Comment by kgilmer** [comment]
Hi @gogsbread ,

Yes, that's exactly what I would do.  You may find it better to start with a "leaf" dependency such as `xrescat` rather than `regolith-desktop`.  I'm guessing that once one package is building via RPM then the rest will follow relatively quickly.  Here is some dep graphs that might help visualize the package porting work.  Top level:

![rd](https://user-images.githubusercontent.com/49683/193715041-2f5f97ea-dc59-48fb-ab32-c5a5726ce720.png)

`regolith-session-flashback` (the X11 specific package of Regolith):

![rd](https://user-images.githubusercontent.com/49683/193716028-8f538c81-ee89-406b-9379-b8f4a3a1172c.png)

Note these graphs I'm running from my dev laptop which has some unreleased stuff.  The graphs are meant to provide more of a "big picture" than provide specific package dependency info.

**Reply by MartinMorlot** [reply]
Thanks for this, I am looking into building Regolith-Sway-session for Archlinux and this is immensely helpful for this purpose. I know someone once ported the project for i3, but it seems he kinda dropped the ball on this lately (which I am absolutely not complaining about as I kknow it takes time and is on a volunteer basis)

Maybe you also have the graph for regolith-sway-session?

Thanks.
